tools/cli/go-whisk-cli/commands/api.go

```diff
@@ -256,37 +259,35 @@ var apiDeleteCmd = &cobra.Command{
             return whiskErr
         }
 
-        api := new(whisk.Api)
-        options := new(whisk.ApiOptions)
-        options.Force = true
-
+        apiDeleteReq := new(whisk.ApiDeleteRequest)
+        apiDeleteReqOptions := new(whisk.ApiDeleteRequestOptions)
         // Is the argument a basepath (must start with /) or an API name
         if _, ok := isValidBasepath(args[0]); !ok {
             whisk.Debug(whisk.DbgInfo, "Treating '%s' as an API name; as it does not begin with '/'\n", args[0])
-            options.ApiBasePath = args[0]
+            apiDeleteReqOptions.ApiBasePath = args[0]
         } else {
-            options.ApiBasePath = args[0]
+            apiDeleteReqOptions.ApiBasePath = args[0]
         }
 
         if (len(args) > 1) {
             // Is the API path valid?
             if whiskErr, ok := isValidRelpath(args[1]); !ok {
                 return whiskErr
             }
-            options.ApiRelPath = args[1]
+            apiDeleteReqOptions.ApiRelPath = args[1]
         }
         if (len(args) > 2) {
             // Is the API verb valid?
             if whiskErr, ok := IsValidApiVerb(args[2]); !ok {
                 return whiskErr
             }
-            options.ApiVerb = strings.ToUpper(args[2])
+            apiDeleteReqOptions.ApiVerb = strings.ToUpper(args[2])
         }
 
-        _, err := client.Apis.Delete(api, options)
+        _, err := client.Apis.Delete(apiDeleteReq, apiDeleteReqOptions)
         if err != nil {
-            whisk.Debug(whisk.DbgError, "client.Apis.Delete(%s) error: %s\n", api.Id, err)
-            errMsg := wski18n.T("Unable to delete action: {{.err}}", map[string]interface{}{"err": err})
+            whisk.Debug(whisk.DbgError, "client.Apis.Delete(%#v, %#v) error: %s\n", apiDeleteReq, apiDeleteReqOptions, err)
+            errMsg := wski18n.T("Unable to delete API: {{.err}}", map[string]interface{}{"err": err})
```

Comment:
Because the API being deleted can be specified as the entire API, an base path + relative path, or base path + relative path + operation, I kept a simple error message that handles all API specification cases.